frontend/src/modules/EditZing/Components/EmailModal.tsx

```diff
@@ -40,20 +49,100 @@ export const EmailModal = ({
   )
 
   // template editor logic
-  const [selectedTemplate, setSelectedTemplate] = useState<
-    EmailTemplate | undefined
-  >(undefined)
+  const [selectedTemplate, setSelectedTemplate] = useState<EmailTemplate>(
+    filteredTemplates[0]
+  )
 
   // Special value substitution in template HTML
   const replaceMap = {
     '{{COURSE_NAME}}': courseNames.join('/'),
   }
+
   const replacedHtml = Object.entries(replaceMap).reduce(
     (prev, [key, value]) => prev.replaceAll(key, value),
     selectedTemplate?.html || ''
   )
 
+  /** Replaces {{NEW_STUDENT_NAME}} and {{OTHER_STUDENTS_NAMES}} with their proper values
+   * in the email template. {replacedHtml} is a general replacement common to all groups, so
+   * for this function, we must pass in the specific groupNumber and find the array of
+   * group members' names using that, since this information is not available from {EmailModalProps}
+   *
+   * @param groupNumber
+   */
+  const studentNamesHtml = (groupNumber: number) => {
+    const group = courses
+      .find((course: Course) => course.courseId === courseId)
+      ?.groups.find((group: Group) => group.groupNumber === groupNumber)
+
+    if (!group) {
+      return replacedHtml
+    }
+
+    const groupSize = group.members.length
+
+    const members = group.members.map(
+      (member) =>
+        students.find((student) => student.email === member)?.name || ''
+    )
```

Comment:
Also I forgot to remind you to change this to the students' actual names rather than netids.